pages/src/navigation/header/header-search.ts

```diff
@@ -19,13 +20,39 @@ export class ESLDemoSearchBox extends ESLToggleable {
 
   @prop() public closeOnOutsideAction = true;
 
+  private isSearchScriptLoaded: boolean = false;
+
   public onShow(params: ToggleableActionParams): void {
     CSSClassUtils.add(this, this.postCls);
+    this.onShowActions(params);
+  }
+
+  private onShowActions(params: ToggleableActionParams): void {
+    if (this.isSearchScriptLoaded) {
+      this.showSearchElements(params);
+    } else {
+      loadSearchScript().then(() => {
+        this.isSearchScriptLoaded = true;
+        this.initSearchScript(params);
+      });
+    }
+  }
+
+  private initSearchScript(params: ToggleableActionParams): void {
+    (window as any).__gcse = {
+      parsetags: 'onload',
+      initializationCallback: (): void => this.showSearchElements(params)
+    };
+  }
+
+  private showSearchElements(params: ToggleableActionParams): void {
     afterNextRender(() => super.onShow(params));
     if (this.autofocus) {
       const $focusEl = TraversingQuery.first(this.firstFocusable, this) as HTMLElement;
       $focusEl && window.setTimeout(() => $focusEl.focus(), parseNumber(this.postClsDelay));
     }
+    const loadingAnimationEL = this.querySelector('.animation-loading')!;
```

Comment:
Maybe I missed up something, what means '!' at the end of sentence?

Reply:
`!` at the end of the object marks it any for typescript

Reply:
But the same as in the point below, it's better properly check the null result instead of disabling types